backend/schema/metadataretry.go

```diff
@@ -0,0 +1,124 @@
+package schema
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+	"google.golang.org/protobuf/proto"
+
+	schemapb "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1/schema"
+)
+
+const (
+	MinBackoffLimitStr = "1s"
+	MinBackoffLimit    = 1 * time.Second
+	MaxBackoffLimitStr = "1d"
+	MaxBackoffLimit    = 24 * time.Hour
+)
+
+type MetadataRetry struct {
+	Pos Position `parser:"" protobuf:"1,optional"`
+
+	Count      *int   `parser:"'+' 'retry' (@Number Whitespace)?" protobuf:"2,optional"`
+	MinBackoff string `parser:"@(Number Ident)?" protobuf:"3"`
+	MaxBackoff string `parser:"@(Number Ident)?" protobuf:"4"`
```

Comment:
FYI, the reasoning behind `@(Number Ident)?` is that I could not find a sane way to parse:
- multiple number + unit pairs while ensuring no whitespace inbetween
- `1m30s` is tokenized as `1` (number) + `m30s` (ident)
Instead we are just getting these in as strings and then parsing them ourselves using regex. This helps us customize the error messages for more cases.